src/coreclr/jit/fginline.cpp

```diff
@@ -586,8 +586,70 @@ class SubstitutePlaceholdersAndDevirtualizeWalker : public GenTreeVisitor<Substi
                 }
 
                 CORINFO_CONTEXT_HANDLE contextInput = context;
+                context                             = nullptr;
                 m_compiler->impDevirtualizeCall(call, nullptr, &method, &methodFlags, &contextInput, &context,
                                                 isLateDevirtualization, explicitTailCall);
+                if (context != nullptr && !m_compiler->compCurBB->HasFlag(BBF_INTERNAL))
+                {
+                    CORINFO_METHOD_INFO methInfo;
+                    if (m_compiler->info.compCompHnd->getMethodInfo(call->gtCallMethHnd, &methInfo, context) &&
+                        genActualType(JITtype2varType(methInfo.args.retType)) == genActualType(call->TypeGet()))
+                    {
+                        IL_OFFSET ilOffset = 0;
+                        INDEBUG(ilOffset = call->gtRawILOffset);
```

Comment:
Seems like we should always set the il offset to the call's IL offset?

Reply:
Isn't it debug only (used for printing the inlining tree)?

Reply:
We always set it for regular inlining, so we should do the same here.